src/util.rs

```diff
@@ -345,6 +345,52 @@ pub unsafe fn get_gc_userdata<T: Any>(state: *mut ffi::lua_State, index: c_int)
     ud
 }
 
+pub unsafe fn init_userdata_metatable_index(
+    state: *mut ffi::lua_State
+) -> Result<()> {
+    match ffi::luaL_dostring(state, cstr!(r#"return function (__index, field_getters, methods)
+        return function (self, key)
+            if field_getters ~= nil and rawget(field_getters, key) ~= nil then
+                return field_getters[key](self)
+            elseif methods ~= nil and methods[key] ~= nil then
+                return methods[key]
+            elseif type(__index) == 'table' then
+                return rawget(__index, key)
+            elseif type(__index) == 'function' then
+                return __index(self, key)
+            else
+                error(string.format('attempt to get an unknown field \'%s\'', key))
```

Comment:
`string` module can simply be not available or be not a standard module.
Strictly speaking we cannot rely on `type` and `rawset` methods as well.

Reply:
+error too